.github/actions/with-docker/action.yml

```diff
@@ -8,34 +8,46 @@ inputs:
     description: 'Subdirectory where code is cloned.'
     required: false
     default:
-  distro:
-    description: 'Distribution to setup Docker for.'
+  github-app-id:
+    description: 'GitHub App ID for token generation'
     required: true
+    type: string
+  github-app-private-key:
+    description: 'GitHub App private key for token generation'
+    required: true
+    type: string
   llvm-version:
     description: 'LLVM version to use.'
     required: true
 runs:
   using: 'composite'
   steps:
+  - uses: actions/create-github-app-token@v1
+    id: temp-token
+    name: Generate temporary GitHub Token
+    with:
+      app-id: "${{ inputs.github-app-id }}"
+      private-key: "${{ inputs.github-app-private-key }}"
+      owner: "${{ github.repository_owner }}"
   - name: 'Set up Docker'
     shell: bash {0}
     env:
       TAG_NAME: ${{ inputs.tag }}
       SUBDIR: ${{ inputs.subdir }}
-      BASE_IMAGE: ${{ inputs.distro }}
       LLVM_VERSION: ${{ inputs.llvm-version }}
     run: |
       set -euxo pipefail
 
       K_VERSION=$(cat ${SUBDIR}deps/k_release)
-      Z3_VERSION=$(cat ${SUBDIR}deps/z3_version)
+      GITHUB_ACTOR=${{ github.actor }}
+      GITHUB_TOKEN=${{ steps.temp-token.outputs.token }}
 
       docker build . --file ${SUBDIR}.github/workflows/Dockerfile    \
-          --tag runtimeverification/${TAG_NAME}                      \
-          --build-arg BASE_IMAGE=${BASE_IMAGE}                       \
-          --build-arg Z3_VERSION=${Z3_VERSION}                       \
+          --tag pi2/${TAG_NAME}                                      \
           --build-arg LLVM_VERSION=${LLVM_VERSION}                   \
           --build-arg K_VERSION=${K_VERSION}                         \
+          --build-arg GITHUB_TOKEN="${GITHUB_TOKEN}" \
+          --build-arg GITHUB_ACTOR="${GITHUB_ACTOR}" \
```

Comment:
These build arguments seem to be unreferenced in the `Dockerfile`.

If I'm right, I would either add a command that references them or remove these build arguments. If the build arguments are removed, I believe the following could also be removed:

1. the token generation step in the file above
2. the declared input parameters `github-app-id` and `github-app-private-key`
3. the corresponding passed in arguments to the `with-docker/action.yml` command in the `test-pr.yml` file